app/models/audit_trail.rb

```diff
@@ -71,4 +32,33 @@ def changed_by_user
   def marked_urgent?
     modified.include?('urgent_flag') && modified['urgent_flag'] == true
   end
+
+  def shaped_changes
```

Comment:
This is the most significant change here -- rather than `changed_from` and `changed_to`, this rolls it all up into one big hash, because it's easier to keep them aligned if they're in the same column.